File: eng_activator_app/lib/models/dictionary/dictionary_response.dart
import 'package:eng_activator_app/models/dictionary/dictionary_entry.dart';
import 'package:eng_activator_app/shared/functions.dart';

class DictionaryResponse {
  late List<DictionaryEntry> dictionaryEntries = [];
  late List<String> recommendations = [];

  DictionaryResponse({
    required List<DictionaryEntry> dictionaryEntries,
    required List<String> recommendations,
  }) {
    this.dictionaryEntries = dictionaryEntries;
    this.recommendations = recommendations;
  }

  DictionaryResponse.fromJson(Map<String, dynamic> json) {
    dictionaryEntries = Converter.convertToList(json, "dictionaryEntries", (e) => DictionaryEntry.fromJson(e));
    recommendations = Converter.toList(json, "recommendations");
  }
}

File: EngActivator/EngActivator.APP/Dtos/Dictionary/DictionaryResponse.cs
using System.Collections.Generic;

namespace EngActivator.APP.Dtos
{
    public class DictionaryResponse
    {
        public List<DictionaryEntry> DictionaryEntries { get; set; } = new List<DictionaryEntry>();
        public List<string> Recommendations { get; set; } = new List<string>();
    }
}
